components/button/index.css

```diff
@@ -147,6 +147,11 @@ governing permissions and limitations under the License.
     box-shadow: none;
   }
 
+  &:active {
+    /* stylelint-disable-next-line spectrum-tools/no-unknown-custom-properties */
+    transform: perspective(max(var(--spectrum-downstate-height), var(--spectrum-downstate-width) * var(--spectrum-component-size-width-ratio-down))) translateZ(var(--spectrum-component-size-difference-down));
```

Comment:
If we're expecting users to dynamically fetch this width/height data, will we need to namespace these?

Reply:
Any use cases you have in mind where a user will do it from their end?

Reply:
Originally when we were exploring a class-based approach, users were be expected to set these width and height variables in their implementation. But if we go with the component-based approach shown in this PR, we may be able to set these and ship them as part of the CSS instead.

Reply:
Namespacing might be necessary if there are cases where a component with a down state is nested within another component that has a down state? I'm not sure.

Maybe we could use --mod naming on these? `--mod-button-downstate-height`?